src/main/java/org/springframework/data/couchbase/repository/query/support/N1qlUtils.java

```diff
@@ -75,14 +86,40 @@ public static Expression escapedBucket(String bucketName) {
    * stored in Couchbase. Notably it will select the content of the document AND its id and cas.
    *
    * @param bucketName the bucket that stores the entity documents (will be escaped).
+   * @param returnedType Returned type projection information from result processor.
    * @return the needed SELECT clause of the statement.
    */
-  public static FromPath createSelectClauseForEntity(String bucketName) {
+  public static FromPath createSelectClauseForEntity(String bucketName, ReturnedType returnedType) {
     Expression bucket = escapedBucket(bucketName);
     Expression metaId = path(meta(bucket), "id").as(CouchbaseOperations.SELECT_ID);
     Expression metaCas = path(meta(bucket), "cas").as(CouchbaseOperations.SELECT_CAS);
+    List<Expression> expList = new ArrayList<Expression>();
+    expList.add(metaId);
+    expList.add(metaCas);
+
+    if (returnedType.isProjecting()) {
+      List<String> properties = returnedType.getInputProperties();
```

Comment:
I guess this could be simplified to:

```java
CouchbasePersistentEntity<?> entity = mappingContext.getPersistentEntity(returnedType.getDomainType());

for (String  property : properties) {
  expList.add(path(bucket, entity.getPersistentProperty(property).getFieldName()));
}
```

Creating a `MappingContext` from scratch is suboptimal here. Is there a way we can reuse an existing instance already available? Generally speaking I wonder why the execution here has to do the property-to-field-name-mapping. Shouldn't that be the responsibility of the query execution in `CouchbaseTemplate`?